SCF-Assignment-4/ArrayOperations/ArrOperation.java

```diff
@@ -0,0 +1,242 @@
+package ArrayOperations;
+
+public class ArrOperation {
+	// Part1
+	/**
+	 * This method calculate the length of largest mirror in array
+	 * 
+	 * @param array
+	 *            input array
+	 * @return size of largest mirror section in array
+	 */
+	public static int calculateLargestMirror(int[] array) {
+		int size = array.length - 1;
+		if (size == 0) {
+			throw new AssertionError("enter valid array");
+		}
+		int j = size;
+		int i = 0;
+		int count = 0;
+		int maxLength = 0;
+		while (i < size) {
+			while (j >= 0) {
+				if (array[i] == array[j]) {
+					i++;
+					count++;
+				}
+				j--;
+			}
+			if (maxLength < count) {
+				maxLength = count;
+				count = 0;
+			}
+			i++;
+		}
+		return maxLength;
+	}
+
+	// Part 2
+	/**
+	 * This method calculates the number of clumps in input array
+	 * 
+	 * @param array
+	 *            input array
+	 * @return number of clumps in input array
+	 */
+	public static int calculatearrayClumps(int[] array) {
+		int size = array.length;
+		if (size == 0) {
+			throw new AssertionError("enter valid array");
+		}
+		int count = 0;
+		int numOfClumps = 0;
+		for (int i = 0; i < size - 1; i++) {
+			if (array[i] == array[i + 1]) {
+				count++;
+			} else {
+				if (count > 0) {
+					numOfClumps++;
+				}
+				count = 0;
+			}
+		}
+		return numOfClumps;
+	}
+
+	// Part 3
+
+	/**
+	 * This method calculates the count of x in array
+	 * 
+	 * @param x
+	 *            number whose successor needs to be fixed
+	 * @param array
+	 *            input array
+	 * @param size
+	 *            size of input array
+	 * @return number of x
+	 */
+	private static int numOfX(int x, int[] array, int size) {
+		int count = 0;
+		for (int i = 0; i < size; i++) {
+			if (array[i] == x) {
+				count++;
+			}
+		}
+		return count;
+	}
+
+	/**
+	 * This method calculates the count of y in array
+	 * 
+	 * @param y
+	 *            number which needs to be fixed as successor

```

Comment:
The description here should be like:
" Number whose number of occurrences need to find"